hq/app/controllers/SnykController.scala

```diff
@@ -0,0 +1,75 @@
+package controllers
+
+
+import auth.SecurityHQAuthActions
+import play.api.Configuration
+import play.api.libs.ws.WSClient
+import play.api.mvc._
+
+import scala.concurrent.ExecutionContext
+import logic.SnykDisplay
+import api.Snyk
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagement
+import com.gu.configraun.Errors.ConfigraunError
+import utils.attempt.{Attempt, FailedAttempt, Failure}
+import utils.attempt.PlayIntegration.attempt
+import model._
+
+class SnykController(val config: Configuration)
+                    (implicit
+                     val ec: ExecutionContext,
+                     val wsClient: WSClient,
+                     val bodyParser: BodyParser[AnyContent],
+                     val controllerComponents: ControllerComponents,
+                     val assetsFinder: AssetsFinder,
+                     val awsClient: AWSSimpleSystemsManagement,
+                     val configraun: com.gu.configraun.models.Configuration)
+  extends BaseController  with SecurityHQAuthActions
+{
+
+  def snyk() = {
+
+    Action.async {
+      attempt {
+        for {
+          token <- getToken
+          organisation <- getOrganisation
+
+          organisationResponse <- Snyk.getSnykOrganisations(token, wsClient)
+          organisationId <- SnykDisplay.getOrganisationId(organisationResponse.body, organisation)
+
+          projectResponse <- Snyk.getProjects(token, organisationId, wsClient)
+          projects <- SnykDisplay.getProjectIdList(projectResponse.body)
+
+          vulnerabilitiesResponse <- Snyk.getProjectVulnerabilities(organisationId, projects, token, wsClient)
+          vulnerabilitiesResponseBodies = vulnerabilitiesResponse.map(a => a.body)
+          parsedVulnerabilitiesResponse <- SnykDisplay.parseProjectVulnerabilities(vulnerabilitiesResponseBodies)
+
+          results = SnykDisplay.labelProjects(projects, parsedVulnerabilitiesResponse)
+          sortedResult = SnykDisplay.sortProjects(results)
+        } yield Ok(views.html.snyk.snyk(sortedResult))
+      }
+    }
+
+  }
+
+  def getToken = configraun.getAsString("/snyk/token") match {
```

Comment:
Not sure these belong in the controller (they aren't actions), but not the end of the world if you really want them here.

Reply:
I've also noticed that other configraun happens at start time. That seems very reasonable, why aren't these done the same way?

Reply:
The stack/stage/app work happens at start time because configraun needs them to initialise.

I would usually rather have the snyk-specific code in a snyk-specific class, but I notice that the SnykController is only instantiated when /snyk is requested, so perhaps it would be better to force it up front in the AppController.

Discuss?  / Confirm? pls